drivers/i3c/master/mipi-i3c-hci/dat_v1.c

```diff
@@ -141,13 +141,24 @@ static void hci_dat_v1_free_entry(struct i3c_hci *hci, unsigned int dat_idx)
 static void hci_dat_v1_set_dynamic_addr(struct i3c_hci *hci,
 					unsigned int dat_idx, u8 address)
 {
+#ifdef CONFIG_ARCH_ASPEED
+	if (dat_idx != address) {
+		int ret;
+
+		ret = hci_dat_v1_alloc_entry(hci, address);
+		if (ret < 0)
+			dev_err(&hci->master.dev, "Allocate entry: %d", ret);
+		hci_dat_v1_free_entry(hci, dat_idx);
```

Comment:
should this be freed only if there is an error? if thats the case, braces required at line 149 and and 151